mParticle-Apple-SDK/Utils/MPLaunchInfo.swift

```diff
@@ -0,0 +1,103 @@
+
+    @objc(initWithURL:sourceApplication:annotation:) public init?(url: URL?, sourceApplication: String?, annotation: Any?) {
+        guard let url else { return nil }
+
+        self.sourceApp = sourceApplication
+        self.url = url
+        super.init()
+        
+        if let sourceApp = self.sourceApp {
+            let urlString = url.absoluteString
+            let appLinksRange = urlString.range(of: "al_applink_data")
+            self.sourceApplication = appLinksRange == nil ? sourceApp : "AppLink(\(sourceApp))"
+        } else {
+            self.sourceApplication = nil
+        }
+        
+        self.annotation = convertAnnotationToString(rawAnnotation: annotation)
+        
+        self.options = [:]
+        if self.sourceApplication != nil {
+            self.options?[UIApplication.OpenURLOptionsKey.sourceApplication.rawValue] = self.sourceApplication
+        }
+        if self.annotation != nil {
+            self.options?[UIApplication.OpenURLOptionsKey.annotation.rawValue] = self.annotation
+        }
+    }
+
+    @objc(initWithURL:options:) public init?(url: URL?, options: [String : Any]? = nil) {
+        guard let url else { return nil }
+
+        self.sourceApp = options?[UIApplication.OpenURLOptionsKey.sourceApplication.rawValue] as? String
+        self.annotation = options?[UIApplication.OpenURLOptionsKey.annotation.rawValue] as? String
+        self.url = url
+        super.init()
+                
+        if let sourceApp = self.sourceApp {
+            let urlString = url.absoluteString
+            let appLinksRange = urlString.range(of: "al_applink_data")
+            self.sourceApplication = appLinksRange == nil ? sourceApp : "AppLink(\(sourceApp))"
+        } else {
+            self.sourceApplication = nil
+        }
+    }
+    
+    private func convertAnnotationToString(rawAnnotation: Any?) -> String? {
+        guard let rawAnnotation else { return nil }
+        
+        if let annotationString = rawAnnotation as? String {
+            return annotationString
+        } else if let annotationDictionary = rawAnnotation as? [String : Any] {
+            var finalDict: [String : String] = [:]
+            for (key, value) in annotationDictionary {
+                if let valueString = self.convertAnnotationToString(rawAnnotation: value) {
+                    finalDict[key] = valueString
+                }
+            }
+            if finalDict.isEmpty { return nil } else {
+                return serializeObject(finalDict)
+            }
+        } else if let annotationArray = rawAnnotation as? [Any] {
+            var finalArray: [String] = []
+            for value in annotationArray {
+                if let valueString = self.convertAnnotationToString(rawAnnotation: value) {
+                    finalArray.append(valueString)
+                }
+            }
+            if finalArray.isEmpty { return nil } else {
+                return serializeObject(finalArray)
+            }
```

Comment:
```suggestion
            if finalArray.isEmpty { 
                return nil 
            } else {
                return serializeObject(finalArray)
            }
```

I'm all for the single line guard statements, but I think should be on separate lines for readability.

Reply:
```suggestion
            return finalArray.isEmpty ? nil : serializeObject(finalArray)
```

Or actually even better, why not a ternary operator (if Swift allows it) in this case?

Reply:
tested to confirm and yeah works fine